pkg/provider/acme/provider.go

```diff
@@ -806,6 +914,28 @@ func (p *Provider) getValidDomains(ctx context.Context, domain types.Domain) ([]
 	return cleanDomains, nil
 }
 
+// certExists returns whether a certificate exists for the given domain.
+func (p *Provider) certExists(ctx context.Context, domain types.Domain) bool {
+	validDomains, err := p.getValidDomains(ctx, domain)
+	if err != nil {
+		return false
+	}
+
+	if len(validDomains) > 1 {
+		domain = types.Domain{Main: validDomains[0], SANs: validDomains[1:]}
+	} else {
+		domain = types.Domain{Main: validDomains[0]}
+	}
+
+	for _, cert := range p.certificates {
+		if reflect.DeepEqual(domain, cert.Certificate.Domain) {
+			return true
+		}
+	}
```

Comment:
Will this work correctly if e.g. the SANs in a certificate are in a different order than what `p.getValidDomains()` happens to return?